src/main/java/com/google/ehub/servlets/ItemPageServlet.java

```diff
@@ -42,14 +44,19 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       Gson gson = new Gson();
       response.setContentType("application/json");
       response.getWriter().println(gson.toJson(comments));
+      
  }
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    long itemId = Long.parseLong(request.getParameter("itemId"));
+    Optional<EntertainmentItem> optionalItem = EntertainmentItemDatastore.getInstance().queryItem(itemId);
+    EntertainmentItem item = optionalItem.get();
+    System.out.println("Item:" + item);
```

Comment:
You should check that the optional is not empty before getting the object that it contains.